lib/charms/mysql/v0/mysql.py

```diff
@@ -1579,6 +1659,37 @@ def get_cluster_primary_address(
 
         return matches.group(1)
 
+    def get_primary_label(self) -> Optional[str]:
+        """Get the label of the cluster's primary."""
+        status = self.get_cluster_status()
+        if not status:
+            return None
+        for label, value in status["defaultreplicaset"]["topology"].items():
+            if value["memberrole"] == "primary":
+                return label
+
+    def set_cluster_primary(self, new_primary_address: str) -> None:
+        """Set the cluster primary.
+
+        Args:
+            new_primary_address: Address of node to set as cluster's primary
+
+        Raises:
+            MySQLSetClusterPrimaryError: If the cluster primary could not be set
+        """
+        logger.debug(f"Setting cluster primary to {new_primary_address}")
+
+        set_cluster_primary_commands = (
+            f"shell.connect_to_primary('{self.server_config_user}:{self.server_config_password}@{self.instance_address}')",
+            f"cluster = dba.get_cluster('{self.cluster_name}')",
+            f"cluster.set_primary_instance('{new_primary_address}')",
```

Comment:
Q: @paulomach why don't we use a single mysql-shell call?
```
f"dba.get_cluster('{self.cluster_name}').set_primary_instance('{new_primary_address}')"
```

IMHO Pros: 2x faster (no unnecessary network roundtrips) and less code.
Cons: long Python line (and less readability) :-)

Reply:
speed wise, i don't think it changes, since you're still doing two API calls.
i do like chaining methods though, even from readability point of view.
I'm taking a note to implement on the next lib change

Reply:
https://warthogs.atlassian.net/browse/DPE-2399